samples/basic/fade_led/src/main.c

```diff
@@ -14,46 +14,57 @@
 #include <zephyr/device.h>
 #include <zephyr/drivers/pwm.h>
 
-static const struct pwm_dt_spec pwm_led0 = PWM_DT_SPEC_GET(DT_ALIAS(pwm_led0));
+#define PWM_LED_ALIAS(i) DT_ALIAS(_CONCAT(pwm_led, i))
+#define PWM_LED(i, _)                                                                              \
+	IF_ENABLED(DT_NODE_EXISTS(PWM_LED_ALIAS(i)), (PWM_DT_SPEC_GET(PWM_LED_ALIAS(i)),))
+
+/* Support up to 4 LEDs*/
+static const struct pwm_dt_spec pwm_leds[] = {LISTIFY(4, PWM_LED, ())};
```

Comment:
Why limit up to 4?
I don't think there's any problem with not placing any limitation here.

Reply:
I can raise the number but there needs to be some cap because there needs to be some sort of iteration over the LEDs.

Reply:
You can use the DT macros to run for each entry and should be able to construct any length array from that

Reply:
Can you give an example? In this case there is no matching compatible string for a pwm LED or something like that so I don't understand how this would be done.